tests/test_parser.py

```diff
@@ -1782,3 +1782,153 @@ def test_redirect_error_to_output(r, o):
     assert check_xonsh_ast({}, '$[echo "test" {} {}> test.txt]'.format(r, o), False)
     assert check_xonsh_ast({}, '$[< input.txt echo "test" {} {}> test.txt]'.format(r, o), False)
     assert check_xonsh_ast({}, '$[echo "test" {} {}> test.txt < input.txt]'.format(r, o), False)
+
+# test invalid expressions
+
+def test_syntax_error_del_literal():
+    with pytest.raises(SyntaxError):
+        PARSER.parse('del 7')
+
+def test_syntax_error_del_constant():
+    with pytest.raises(SyntaxError):
+        PARSER.parse('del True')
+
+def test_syntax_error_del_emptytuple():
+    with pytest.raises(SyntaxError):
+        PARSER.parse('del ()')
+
+def test_syntax_error_del_call():
+    with pytest.raises(SyntaxError):
+        PARSER.parse('del foo()')
+
+def test_syntax_error_del_lambda():
+    with pytest.raises(SyntaxError):
+        PARSER.parse('del lambda x: "yay"')
+
+def test_syntax_error_del_ifexp():
+    with pytest.raises(SyntaxError):
+        PARSER.parse('del x if y else z')
+
+def test_syntax_error_del_comps():
+    with pytest.raises(SyntaxError):
+        PARSER.parse('del [i for i in foo]')
+    with pytest.raises(SyntaxError):
+        PARSER.parse('del {i for i in foo}')
+    with pytest.raises(SyntaxError):
+        PARSER.parse('del {k:v for k,v in d.items()}')
+    with pytest.raises(SyntaxError):
+        PARSER.parse('del (i for i in foo)')
+
```

Comment:
you can also use the `@pytest.mark.parametrize` decorator feature which is very handy, just for future reference!